### PR TITLE
Register shape functions for some image related ops

### PR DESCRIPTION
**Pull request by Laurawly**
Added shape functions for image.resize, image.grid_sample and image.affine_grid

@icemelon9 Could you please review?

### CODE REVIEW

File: python/tvm/relay/op/image/_image.py [hunk 1]
```diff
@@ -40,6 +40,33 @@ def compute_resize(attrs, inputs, out_type):
...
+    height_axis = width_axis = channel_axis = 1
+    for i, letter in enumerate(layout):
+        if letter == "H":
+            height_axis = i
+        if letter == "W":
+            width_axis = i
+        if letter == "C":
+            channel_axis = i
+    size = get_const_tuple(attrs.size)
+    return [_resize_func(inputs[0], convert(size), convert(height_axis),
```
Comment (icemelon9) [inline python/tvm/relay/op/image/_image.py:67]: Could you also provide batch axis to the shape func as well?

File: python/tvm/relay/op/image/_image.py [hunk 2]
```diff
@@ -40,6 +40,33 @@ def compute_resize(attrs, inputs, out_type):
...
+def _resize_func(image_shape, size, height_axis, width_axis, channel_axis):
+    out = output_tensor((4,), "int64")
+    out[0] = int64(image_shape[0])
+    out[height_axis] = int64(size[0])
+    out[width_axis] = int64(size[1])
+    out[channel_axis] = image_shape[channel_axis]
+    return out
+
+@reg.register_shape_func("image.resize", False)
+def resize_func(attrs, inputs, _):
```
Comment (icemelon9) [inline python/tvm/relay/op/image/_image.py:53]: change the name to `resize_shape_func`

File: python/tvm/relay/op/image/_image.py [hunk 3]
```diff
@@ -40,6 +40,33 @@ def compute_resize(attrs, inputs, out_type):
 
 reg.register_injective_schedule("image.resize")
 
+@script
+def _resize_func(image_shape, size, height_axis, width_axis, channel_axis):
```
Comment (icemelon9) [inline python/tvm/relay/op/image/_image.py:44]: change the name to `_resize_shape_func`.

File: python/tvm/relay/op/image/_image.py [hunk 4]
```diff
@@ -109,6 +136,23 @@ def compute_affine_grid(attrs, inputs, out_dtype):
 
 reg.register_injective_schedule("image.affine_grid")
 
+@script
+def _affine_grid_func(data, target_shape):
+    out = output_tensor((4,), "int64")
+    out[0] = int64(data[0])
+    out[1] = int64(2)
+    out[2] = int64(target_shape[0])
```
Comment (icemelon9) [inline python/tvm/relay/op/image/_image.py:144]: do we support different layout for `affine_grid`?

File: python/tvm/relay/op/image/_image.py [hunk 5]
```diff
@@ -118,3 +162,19 @@ def compute_grid_sample(attrs, inputs, out_dtype):
...
 
 reg.register_injective_schedule("image.grid_sample")
+
+@script
+def _grid_sample_func(data, grid):
+    out = output_tensor((4,), "int64")
+    out[0] = int64(data[0])
+    out[1] = int64(data[1])
+    out[2] = int64(grid[2])
+    out[3] = int64(grid[3])
```
Comment (icemelon9) [inline python/tvm/relay/op/image/_image.py:172]: same question here. different layout?

**Comment by tqchen** [comment]
ping @Laurawly @icemelon9 please follow up

File: python/tvm/relay/op/image/_image.py (hunk 4 above)
Comment (Laurawly) [reply python/tvm/relay/op/image/_image.py:144]: No I don't think so.

File: python/tvm/relay/op/image/_image.py (hunk 5 above)
Comment (Laurawly) [reply python/tvm/relay/op/image/_image.py:172]: No, I think it only supports NCHW

**Review by icemelon9 — APPROVED** [review]
LGTM. Please resolve the conflict

File: tests/python/relay/test_any.py [hunk 6]
```diff
@@ -891,6 +891,87 @@ def test_any_ndarray_size():
...
+    dtype = "float32"
+    data = relay.var('data', shape=data_shape, dtype=dtype)
+    if layout == "NHWC":
+        size = (data_shape[1] * scale, data_shape[2] * scale)
+    else:
+        size = (data_shape[2] * scale, data_shape[3] * scale)
+    y = relay.image.resize(data, size, layout)
+    mod["main"] = relay.Function([data], y)
+    data_np = np.random.uniform(size=static_data_shape).astype(dtype)
+    for kind in ["debug", "vm"]:
```
Comment (zhiics) [inline tests/python/relay/test_any.py:905]: BTW, this has been moved to `test_result` and you may want to annotate the test with `@tvm.testing.uses_gpu` if gpu is supported

**Review by zhiics — APPROVED** [review]

**Pull request merged**